### PR TITLE
SAR-160 - Readmission measure

### PR DESCRIPTION
**Pull request by anthem123**
# Related Tickets

- [SAR-160](https://jira.amida-tech.com/browse/SAR-160)

This CQL is used to calculate the readmission HEDIS score by checking if a patient had 2 or more unplanned trips to the hospital where they had to stay for more than 1 day within 30 days of each other.

The full list of exceptions is

- Pregnancy and perinatal visits
- Chemotherapy
- Kidney transplant
- Bone marrow transplant
- Organ transplant
- Introduction of pancreatic cells

The final score is

Total readmissions within 30 days / Total applicable admissions

### CODE REVIEW

**Comment by eawest2** [comment]
I don't get the usual logs showing the results of the CQL in the console. Is this intended?

**Review by eawest2 — APPROVED** [review]
LGTM

File: cql/readmission.cql [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+define function getCountWithNull(enc Encounter):
+  case
+    when enc.hospitalization is null then 0
+    when enc.hospitalization.dischargeDisposition is null then 0
+    else Count(enc.hospitalization.dischargeDisposition.coding.code)
+  end
+
+define function hasPregnancyEncounter(eob ExplanationOfBenefit):
+  Exists(([Encounter: "Inpatient Obstetrics NDNQI"] enc where Substring(eob.item[0].encounter[0].reference.value, 9) = enc.id))
```
Comment (jmfadeley) [inline cql/readmission.cql:80]: Check if we could add the Perinatal part with https://vsac.nlm.nih.gov/valueset/2.16.840.1.113762.1.4.1195.334/expansion/Latest

File: cql/readmission.cql [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    and ((enc2.period.end occurs 365 days or less before enc.period.end)
+      or (enc2.period.end occurs 30 days or less after enc.period.end)))
+
+// Step 6 Remove hospital stays if you go to the hospital too much
+define "Final Denominator List":
+  case getInsurance(First("Continuous Enrollment"))
+    when 'Medicare' then if Count("Continuous Enrollment") < 4 then "Continuous Enrollment" else List<ExplanationOfBenefit>{}
+    when 'Medicaid' then if Count("Continuous Enrollment") < 4 then "Continuous Enrollment" else List<ExplanationOfBenefit>{}
+    else if Count("Continuous Enrollment") < 3 then "Continuous Enrollment" else List<ExplanationOfBenefit>{}
+  end
```
Comment (jmfadeley) [inline cql/readmission.cql:98]: Let's create a new ticket to investigate if we need to report a separate outlier ticket. See Step 6 for explanation. Make it a 2 point evaluation ticket and expand to 5 points to solve if needed.

File: cql/readmission.cql [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+// Step 3 - Exlcude various encounters
+define "Numerator Exclusions":
+  [Encounter: "Chemotherapy"]
+    union [Encounter: "Rehabilitation Services"]
+    union [Encounter: "Rehabilitation Therapy"]
+    union [Encounter: "Kidney Transplant"]
+    union [Encounter: "Bone Marrow Transplant"]
+    union [Encounter: "Organ Transplant"]
+    union [Encounter: "Inpatient Obstetrics NDNQI"]
```
Comment (jmfadeley) [inline cql/readmission.cql:133]: Don't forget to add the perinatal here.

File: cql/readmission.cql [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    union [Encounter: "Potentially Planned Procedures"]
```
Comment (jmfadeley) [inline cql/readmission.cql:134]: Make this its own function and check for the "Acute Condition Value Set" to exclude that from the PPP grouping above.

File: cql/readmission.cql (hunk 3 above)
Comment (anthem123) [reply cql/readmission.cql:136]: Perinatal has been added, you have already created pancreatic cell transplant ticket.

File: cql/readmission.cql (hunk 4 above)
Comment (anthem123) [reply cql/readmission.cql]: Acute Condition Value Set has been added.

**Pull request merged**